src/prop.c

```diff
@@ -117,12 +147,148 @@ SEXP prop_(SEXP object, SEXP name) {
           name_sym == sym_properties  ||
           name_sym == sym_abstract  ||
           name_sym == sym_constructor  ||
-          name_sym == sym_validator
-    ))
+          name_sym == sym_validator  ))
       return value;
 
-  error:
+  // Should the constructor always set default prop values on a object instance?
+  // Maybe, instead, we can fallback here to checking for a default value from the
+  // properties list.
 
-  signal_prop_error_unknown_(object, name);
+  signal_prop_error_unknown(object, name);
   return R_NilValue; // unreachable, for compiler
 }
+
+
+static inline
+Rboolean pairlist_contains(SEXP list, SEXP elem) {
+  for (SEXP c = list; c != R_NilValue; c = CDR(c))
+    if (CAR(c) == elem)
+      return TRUE;
+  return FALSE;
+}
+
+static inline
+SEXP pairlist_remove(SEXP list, SEXP elem) {
+  SEXP c0 = NULL, head = list;
+  for (SEXP c = list; c != R_NilValue; c0 = c, c = CDR(c))
+    if (CAR(c) == elem)
+    {
+      if (c0 == NULL)
+        return CDR(c);
+      else
+      {
+        SETCDR(c0, CDR(c));
+        return head;
+      }
+    }
+
+  Rf_warning("Tried to remove non-existent element from pairlist");
+  return R_NilValue;
```

Comment:
Should this just be an error? Probably right? I can't imagine returning `NULL` here would ever really be the right thing to do